dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/DynamoSettings.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.dynamodb
+
+import akka.actor.ActorSystem
+import com.amazonaws.auth._
+import com.typesafe.config.Config
+
+final class DynamoSettings private (
+    val region: String,
+    val host: String,
+    val port: Int,
+    val tls: Boolean,
+    val parallelism: Int,
+    val credentialsProvider: com.amazonaws.auth.AWSCredentialsProvider
+) extends AwsClientSettings {
+
+  require(host.nonEmpty, "A host name must be provided.")
+  require(port > -1, "A port number must be provided.")
+
+  def withRegion(value: String): DynamoSettings = copy(region = value)
+  def withHost(value: String): DynamoSettings = copy(host = value)
+  def withPort(value: Int): DynamoSettings =
+    if (value == 443) {
+      copy(port = value, tls = true)
+    } else copy(port = value)
```

Comment:
I am not sure about these "smart" builders. There could be a case where server expects non-secure connections on port `443` and then it could catch user off-guard. Do we have examples of "smart" builder methods anywhere else?

Reply:
That was my try to keep it a bit more backwards-compatible. The old logic was to switch to https for port 443, only.

Reply:
I removed the 443 magic, existing users need to switch off TLS explicitly.